CI: Travis failed on arm64:  TestSpanStatRaceCondition Fail

CI failure Link:
https://travis-ci.com/github/cilium/cilium/jobs/361628402#L9712

```
panic: Log in goroutine after TestSpanStatRaceCondition/End_function has completed
goroutine 37 [running]:
testing.(*common).logDepth(0x4000156ea0, 0x40000341e0, 0x9a, 0x3)
	/home/travis/.gimme/versions/go1.14.4.linux.arm64/src/testing/testing.go:680 +0x3e8
testing.(*common).log(...)
	/home/travis/.gimme/versions/go1.14.4.linux.arm64/src/testing/testing.go:662
testing.(*common).Errorf(0x4000156ea0, 0x502374, 0x3, 0x4000119800, 0x1, 0x1)
	/home/travis/.gimme/versions/go1.14.4.linux.arm64/src/testing/testing.go:711 +0x6c
github.com/stretchr/testify/assert.Fail(0x5c1560, 0x4000156ea0, 0x40000265a0, 0x11, 0x0, 0x0, 0x0, 0x40815c)
	/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/stretchr/testify/assert/assertions.go:261 +0x228
github.com/stretchr/testify/assert.NotEqual(0x5c1560, 0x4000156ea0, 0x44ff00, 0x9598a0, 0x44ff00, 0x5b3c80, 0x0, 0x0, 0x0, 0x524eb0)
	/home/travis/gopath/src/github.com/cilium/cilium/vendor/github.com/stretchr/testify/assert/assertions.go:650 +0x1dc
github.com/cilium/cilium/pkg/spanstat.TestSpanStatRaceCondition.func8.1(0x4000156ea0, 0x400000e120, 0x4000161380)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/spanstat/spanstat_test.go:185 +0x84
created by github.com/cilium/cilium/pkg/spanstat.TestSpanStatRaceCondition.func8
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/spanstat/spanstat_test.go:184 +0xcc
FAIL	github.com/cilium/cilium/pkg/spanstat	0.033s
FAIL
```

--- a/pkg/spanstat/spanstat_test.go
+++ b/pkg/spanstat/spanstat_test.go
@@ -101,10 +101,11 @@ func (s *SpanStatTestSuite) TestSpanStatSecondsRaceCondition(c *C) {
 	for i := 0; i < 10; i++ {
 		wg.Add(1)
 		go func(span *SpanStat) {
+			defer wg.Done()
 			c.Assert(span1.Seconds(), Not(Equals), float64(0))
 		}(span1)
 	}
-	wg.Done()
+	wg.Wait()
 }
 
 func TestSpanStatRaceCondition(t *testing.T) {
@@ -182,10 +183,11 @@ func TestSpanStatRaceCondition(t *testing.T) {
 			for i := 0; i < 5; i++ {
 				wg.Add(1)
 				go func(span *SpanStat) {
+					defer wg.Done()
 					assert.NotEqual(t, tt.fields.runFunc(span), float64(0))
 				}(span)
 			}
-			wg.Done()
+			wg.Wait()
 		})
 	}
 

